app/uk/gov/hmrc/fileupload/virusscan/ScanningService.scala

```diff
@@ -28,23 +29,54 @@ object ScanningService {
   type ScanResult = Xor[ScanError, ScanResultFileClean.type]
 
   sealed trait ScanError
+
   case object ScanResultVirusDetected extends ScanError
-  case class  ScanResultFailureSendingChunks(ex: Throwable) extends ScanError
+
+  case object ScanReadCommandTimeOut extends ScanError
+
+  case class ScanResultFailureSendingChunks(ex: Throwable) extends ScanError
+
   case object ScanResultUnexpectedResult extends ScanError
-  case class  ScanResultError(ex: Throwable) extends ScanError
+
+  case class ScanResultError(ex: Throwable) extends ScanError
+
   case object ScanResultFileClean
 
   type AvScanIteratee = Iteratee[Array[Byte], Future[ScanResult]]
 
   def scanBinaryData(scanner: () => Iteratee[Array[Byte], Future[ScanResult]],
+                     scanTimeoutAttempts: Int,
                      getFile: (String, String) => Future[QuarantineDownloadResult])
                     (s3KeyAppender: (EnvelopeId, FileId) => String)
-                    (envelopeId: EnvelopeId, fileId : FileId, fileRefId: FileRefId)
+                    (envelopeId: EnvelopeId, fileId: FileId, fileRefId: FileRefId)
                     (implicit ec: ExecutionContext): Future[ScanResult] = {
     val appendedKey = s3KeyAppender(envelopeId, fileId)
-    getFile(appendedKey, fileRefId.value).flatMap {
+    retries(scan(scanner, getFile(appendedKey, fileRefId.value)), scanTimeoutAttempts, fileId)
+  }
+
+  private def retries(scanResult: => Future[ScanResult],
+                      timeoutAttempts: Int,
+                      fileId: FileId,
+                      acc: Int = 1)
```

Comment:
Naming is hard, but I'd be tempted to swap the order of these three arguments, and rename two of them, so its maybe something like:

    fileId: FileId,
    maxTimeoutRetries: Int
    currentTimeoutRetry: Int = 1

`acc` is a bit weird IMO. Its normally used in recursion/folds to build a collection of something, but here we're incrementing a number.

Reply:
Renamed as suggested.